### PR TITLE
Chore: enhance GraphQL

### PR DESCRIPTION
**Pull request by tlgimenes**
## What's the purpose of this pull request?
This PR brings a few enhancements to the GraphQL layer. 
The first enhancement is related to testing dynamic queries. Before, there was not really a way for testing dynamic queries like `search`, `allProducts` etc because we forced all queries to be known from the store's code before executing it (through persisted queries). This increased the security (because we only run known queries), however, made a poor DX. This PR makes it possible to query a full text graphql query during development only, so people can use tools like [GraphQL Playground](https://github.com/graphql/graphql-playground) for building their queries and than use it on their code

The second enhancement is `console.logging` masked errors before they are masked, so users can know exactly what error happened during development. For instance, the console output before and after this PR:
```sh
 ERROR

GraphQLError [Object]: Sorry, something went wrong.
    at maskError (webpack://base.store/./src/api/graphql.ts?:42:14)
    at eval (webpack://base.store/./node_modules/@envelop/core/index.mjs?:1084:67)
    at Array.map (<anonymous>)
    at eval (webpack://base.store/./node_modules/@envelop/core/index.mjs?:1084:54)
    at handleStreamOrSingleExecutionResult

```
The error above is masked and the developer doesn't really know what is happening on the backend. However, after the changes:
```sh
 ERROR

NotFoundError: Not Found: kitchen-and-home-appliancesasd
    at Object.collection
(webpack://base.store/./node_modules/@faststore/api/dist/api.esm.js?:798:11)
    at processTicksAndRejections (node:internal/process/task_queues:93:5) {
  locations: [ { line: 2, column: 3 } ],
  path: [ 'collection' ]
}

Executed function "/api/graphql" in 719ms
```

## How it works? 
Most of the changes were done on two files, `graphql.ts` the function, and `server/index.js`, the core server implementation.

## How to test it?
Make sure nothing has really changed and everything works just fine

### CODE REVIEW

File: src/server/index.js [hunk 1]
```diff
@@ -1,28 +1,92 @@
+/* eslint-disable react-hooks/rules-of-hooks */
 const {
-  getSchema: storeApiGetSchema,
-  getContextFactory: storeApiGetContextFactory,
-} = require('@faststore/api')
+  envelop,
```
Comment (igorbrasileiro) [inline src/server/index.js:3]: Why does this file need to be a js one? Can't do something like this, can?

```ts
// src/server/index.ts
...
export {}
```

```js
// /src/server/index.js
const { } = require('./dist/server/index')
```

File: src/server/index.js [hunk 2]
```diff
@@ -1,28 +1,92 @@
...
+const apiContextFactory = getContextFactory(apiOptions)
+
+const isBadRequestError = (err) => {
+  return err.originalError && err.originalError.name === 'BadRequestError'
+}
 
-const getSchema = () => schema
-const getContextFactory = () => contextFactory
+const formatError = (err) => {
+  console.error(err)
```
Comment (igorbrasileiro) [inline src/server/index.js:34]: Does it make sense to log on production environment? If not, can you remove it if the environment is production?

File: src/api/graphql.ts [hunk 3]
```diff
@@ -1,52 +1,27 @@
...
 
 const parseRequest = (req: GatsbyFunctionRequest) => {
-  const { operationName, variables } =
+  const { operationName, variables, query } =
     req.method === 'POST'
       ? req.body
       : {
           operationName: req.query.operationName,
-          variables: JSON.parse(req.query.variables),
+          variables: JSON.parse(req.query.variables.toString()),
```
Comment (victorhmp) [inline src/api/graphql.ts:11]: Could `req.query.variables` be `undefined` in any case? If so, just not calling `toString()` would pass the `undefined` on and work fine right? How about 

```
variables: JSON.parse(req.query.variables?.toString()),
```

?

File: src/server/index.js [hunk 4]
```diff
@@ -1,28 +1,92 @@
...
+
+  if (err instanceof GraphQLError) {
+    if (!isBadRequestError(err)) {
+      return new GraphQLError('Sorry, something went wrong.')
+    }
+
+    return err
+  }
```
Comment (igorbrasileiro) [inline src/server/index.js:42]:
```suggestion
  if (err instanceof GraphQLError && isBadRequestError(err)) {
    return err
  }
```

Make the code branchless as much as possible.

File: src/api/graphql.ts [hunk 5]
```diff
@@ -57,32 +32,24 @@ const handler = async (
...
+      },
+      { req }
+    )
+
+    if (Array.isArray(response.errors)) {
+      // TODO: Return 400 on userError
+      res.status(500)
     }
 
+    res.setHeader('cache-control', 'no-cache, no-store')
```
Comment (victorhmp) [inline src/api/graphql.ts:52]: Should this header be added in production too?

**Review by victorhmp — CHANGES_REQUESTED** [review]

**Review by igorbrasileiro — APPROVED** [review]

File: src/api/graphql.ts (hunk 5 above)
Comment (tlgimenes) [reply src/api/graphql.ts:52]: Yes, we are not caching graphql data for now

File: src/server/index.js (hunk 1 above)
Comment (tlgimenes) [reply src/server/index.js:3]: it needs to be js because `gatsby-config.js` uses it

File: src/server/index.js (hunk 4 above)
Comment (tlgimenes) [reply src/server/index.js]: nice!

**Review by victorhmp — APPROVED** [review]

File: src/api/graphql.ts [hunk 6]
```diff
@@ -1,52 +1,27 @@
...
 
 const parseRequest = (req: GatsbyFunctionRequest) => {
-  const { operationName, variables } =
+  const { operationName, variables, query } =
     req.method === 'POST'
       ? req.body
       : {
           operationName: req.query.operationName,
-          variables: JSON.parse(req.query.variables),
+          variables: JSON.parse(req.query.variables?.toString()),
```
Comment (emersonlaurentino) [inline src/api/graphql.ts:11]:
```suggestion
          variables: req.query.variables?,
```

Why do you need to transform to string to parse to JSON?

**Review by emersonlaurentino — APPROVED** [review]

File: src/api/graphql.ts (hunk 6 above)
Comment (tlgimenes) [reply src/api/graphql.ts]: I think we don't need it. Nice!

**Pull request merged**